Objects/obmalloc.c

```diff
@@ -1767,7 +1772,7 @@ pymalloc_free(void *ctx, void *p)
      *    nfreepools.
      * 4. Else there's nothing more to do.
      */
-    if (nf == ao->ntotalpools) {
+    if (nf == ao->ntotalpools && ao->nextarena!= NULL) {
```

Comment:
```suggestion
    if (nf == ao->ntotalpools && ao->nextarena != NULL) {
```

Reply:
@methane, thanks for that!  Those are kinds of silly typos I usually don't make .  Gladly accepted the changes :smile:.